File: cash.c
#include <cs50.h>
#include <stdio.h>

int countTheCoins(int cents);

int main(void)
{
    int money;

    // Prompt the user for change owed, in cents
    do
    {
        money = get_int("Change owed:  ");
    }
    while (money <= 0);

    // Calculate how many coins you should give customer
    printf("%i\n", countTheCoins(money));
}

int countTheCoins(int cents)
{
    int number = 0; // holds number of coins needed
    int holder = 0; // for minus operations

    if (cents / 25 != 0)
    {
        number = cents / 25;
        cents = cents - (25 * number); // Subtract the value of those quarters from cents
    }
    if (cents != 0 && cents / 10 != 0)
    {
        number = number + (cents / 10);
        holder = cents / 10;
        cents = cents - (10 * holder);
        holder = 0;
    }
    if (cents != 0 && cents / 5 != 0)
    {
        number = number + (cents / 5);
        holder = cents / 5;
        cents = cents - (5 * holder);
        holder = 0;
    }
    if (cents != 0)
    {
        number = number + cents;
    }
    return number;
}

File: cash.py
from cs50 import get_float

while True:
    dollar = get_float("Change: ")
    if dollar > 0:
        break

change = int(dollar * 100)

number = 0
holder = 0

if change // 25 != 0:
    number = change // 25
    change = change - (25 * number)

if change != 0 and change // 10 != 0:
    number = number + (change // 10)
    holder = change // 10
    change = change - (10 * holder)
    holder = 0

if change != 0 and change // 5 != 0:
    number = number + (change // 5)
    holder = change // 5
    change = change - (5 * holder)
    holder = 0

if change != 0:
    number = number + change

print(number)
